pkg/query-service/migrate/migate.go

```diff
@@ -77,3 +80,89 @@ func Migrate(dsn string) error {
 
 	return nil
 }
+
+func ClickHouseMigrate(conn driver.Conn, cluster string) error {
+
+	database := "CREATE DATABASE IF NOT EXISTS signoz_analytics ON CLUSTER %s"
+
+	localTable := `CREATE TABLE IF NOT EXISTS signoz_analytics.rule_state_history ON CLUSTER %s
+(
+    rule_id LowCardinality(String),
+    rule_name LowCardinality(String),
+    overall_state LowCardinality(String),
+    overall_state_changed Bool,
+    state LowCardinality(String),
+    state_changed Bool,
+    unix_milli Int64 CODEC(Delta(8), ZSTD(1)),
+    fingerprint UInt64 CODEC(ZSTD(1)),
+    value Float64 CODEC(Gorilla, ZSTD(1)),
+    labels String CODEC(ZSTD(5)),
+)
+ENGINE = MergeTree
+PARTITION BY toDate(unix_milli / 1000)
+ORDER BY (rule_id, unix_milli)
+TTL toDateTime(unix_milli / 1000) + toIntervalSecond(15552000)
```

Comment:
keep the ttl configurable ? with default 180 days

Reply:
Not through env or something but I made it configurable. One can always update the default value for `_retention_days` for their desired retention.